Source/ToolCore/Project/Project.cpp

```diff
@@ -172,6 +172,10 @@ bool Project::Load(const String& fullpath)
 
     }
 
+    VariantMap data;
+    data[ProjectBeginLoad::P_PROJECTPATH] = projectFilePath_;
+    data[ProjectBeginLoad::P_PROJECT] = this;
+    SendEvent(E_PROJECTBEGINLOAD, data);
 
```

Comment:
Great, uncouples code and the event can be caught in native, script, or C# ( which isn't script 😉  )